server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServerResourceManager.java

```diff
@@ -480,10 +480,13 @@ private class MemoryManagementFramework {
       tabletReports = Collections.synchronizedMap(new HashMap<>());
       memUsageReports = new LinkedBlockingQueue<>();
       maxMem = context.getConfiguration().getAsBytes(Property.TSERV_MAXMEM);
-      memoryGuardThread = Threads.createThread("Accumulo Memory Guard",
+      // TODO KEVIN RATHBUN guarding against excessive memory usage and initiating minor
+      // compactions are critical tasks of the tablet server. Also, these threads are only created
+      // once.
+      memoryGuardThread = Threads.createCriticalThread("Accumulo Memory Guard",
           OptionalInt.of(Thread.NORM_PRIORITY + 1), this::processTabletMemStats);
       minorCompactionInitiatorThread =
-          Threads.createThread("Accumulo Minor Compaction Initiator", this::manageMemory);
+          Threads.createCriticalThread("Accumulo Minor Compaction Initiator", this::manageMemory);
```

Comment:
guarding against excessive memory usage and initiating minor compactions are critical tasks of the tablet server. Also, these threads are only created once.